mempager/pager.go

```diff
@@ -89,8 +89,12 @@ func (p *Pager) Set(pageNum int, data []byte) {
 
 // growPages will increases the size of the pager's page buffer up till the supplied index
 func (p *Pager) growPages(index int) {
-	for i := len(p.pages); i <= index; i++ {
-		p.pages = append(p.pages, nil)
+	diff := index - len(p.pages)
+	if diff <= 0 {
+		return
+	} else {
```

Comment:
because you just have a return in the if condition, you don't actually need the `else` here - you can just unwrap the code.

Reply:
Sweet, I originally had it that way but wasn't sure if guard clauses like that are normal in go. I prefer without the else too so that's dope